core/src/main/java/com/github/games647/fastlogin/core/shared/FloodgateManagement.java

```diff
@@ -0,0 +1,131 @@
+    protected boolean isRegistered;
+    protected StoredProfile profile;
+    protected String autoLoginFloodgate;
+    protected boolean isLinked;
+    protected boolean performLogin; //will be set to ture if core#run() wasn't interrupted by a return;
+
+    public FloodgateManagement(FastLoginCore<P, C, T> core, P player, FloodgatePlayer floodgatePlayer) {
+        this.core = core;
+        this.player = player;
+        this.floodgatePlayer = floodgatePlayer;
+        this.username = getName(player);
+    }
+
+    @Override
+    public void run() {
+        core.getPlugin().getLog().info(
+                "Player {} is connecting through Geyser Floodgate.",
+                username);
+
+        // check if the Bedrock player is linked to a Java account 
+        isLinked = floodgatePlayer.getLinkedPlayer() != null;
+        AuthPlugin<P> authPlugin = core.getAuthPluginHook();
+
+        autoLoginFloodgate = core.getConfig().get("autoLoginFloodgate").toString().toLowerCase();
+        String autoRegisterFloodgate = core.getConfig().get("autoRegisterFloodgate").toString().toLowerCase();
+        String allowNameConflict = core.getConfig().get("allowFloodgateNameConflict").toString().toLowerCase();
+        
+        try {
+            isRegistered = authPlugin.isRegistered(username);
+        } catch (Exception e) {
+            core.getPlugin().getLog().error(
+                    "An error has occured while checking if player {} is registered",
+                    username);
+            return;
+        }
+
+        //decide if checks should be made for conflicting Java player names
+        if (!isLinked //linked players have the same name as their Java profile
+                // if allowNameConflict is 'false' or 'linked' and the player had a conflicting
+                // name, than they would have been kicked in FloodgateHook#checkNameConflict
+                && allowNameConflict.equals("true") &&
+                (
+                        autoLoginFloodgate.equals("no-conflict")
+                        || !isRegistered && autoRegisterFloodgate.equals("no-conflict"))
+                ) {
+            // check for conflicting Premium Java name
+            Optional<Profile> premiumUUID = Optional.empty();
+            try {
+                premiumUUID = core.getResolver().findProfile(username);
+            } catch (IOException | RateLimitException e) {
+                core.getPlugin().getLog().error(
+                        "Could not check wether Floodgate Player {}'s name conflits a premium Java player's name.",
+                        username);
+                return;
+            }
+
+            //stop execution if player's name is conflicting
+            if (premiumUUID.isPresent()) {
+                return;
+            }
+        }
+
+        if (!isRegistered && autoRegisterFloodgate.equals("false")) {
+            core.getPlugin().getLog().info(
```

Comment:
This could be identified as SPAM. Maybe you could add the player name, so it's addressable.

Reply:
I actually wanted to remove this weeks ago, but forgot to do it.